src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -273,7 +275,12 @@ public static JsonNode convertToJson(Schema schema, Object logicalValue) {
           return JsonNodeFactory.instance.textNode(charSeq.toString());
         case BYTES:
           if (schema != null && Decimal.LOGICAL_NAME.equals(schema.name())) {
-            return JsonNodeFactory.instance.numberNode((BigDecimal) value);
+            BigDecimal bigDecimalValue = (BigDecimal) value;
+            if (bigDecimalValue.precision() > MAX_SNOWFLAKE_NUMBER_PRECISION) {
+              // in order to prevent losing precision, convert this value to text
+              return JsonNodeFactory.instance.textNode(bigDecimalValue.toString());
+            }
+            return JsonNodeFactory.instance.numberNode(bigDecimalValue);
```

Comment:
does this change need to be made for int64 and float64 type too?

Reply:
@wicknicks `int64`(Java's `long`) and `float64`(Java's `double`) don't exceed Snowflake's precision of 38 digits, so no change is needed for these types. I also updated the test to make sure I use max number of significant digits for these types.